cibuildwheel/logger.py

```diff
@@ -164,7 +165,11 @@ def notice(self, message: str) -> None:
             c = self.colors
             print(f"cibuildwheel: {c.bold}note{c.end}: {message}\n", file=sys.stderr)
 
-    def warning(self, message: str) -> None:
+    def warning(self, message: str, *, deduplicate: bool = False) -> None:
```

Comment:
Maybe we could just always do this for warnings?

Reply:
Think I'd prefer to keep it explicit, it's a bit surprising at the call site otherwise. Might also make some unit tests buggy as this is shared global state.

Reply:
^ actually the global state thing might be a reason to do it on Options rather than here.

Reply:
> I guess if it was an instance variable it still wouldn't over trigger?

I don't think so. There's only one instance of `Options` in a normal running process. The only reason I hesitate to do it globally is that I can imagine a unit test expecting to see a warning but depending on test execution order, it sometimes gets triggered somewhere else and the test becomes flaky/hard to debug.

That said it's a hypothetical issue at this point, we can leave it here if you wish.